riscv/encoding.h

```diff
@@ -358,7 +358,9 @@
 #define PMP_A     0x18
 #define PMP_L     0x80
 #define PMP_SHIFT 2
+#define PMP_ADDR_BITS  54
 
+#define PMP_OFF   0x00
```

Comment:
`encoding.h` is an auto-generated file and so shouldn't be manually modified.  Instead of introducing `PMP_OFF` to write `mode == PMP_TOR || mode == PMP_OFF`, you could write `!(mode == PMP_NA4 || mode == NAPOT)`, for example.

Reply:
I'm working on a new PR that's inspired by yours, so no need to make the code changes.  Once I'm finished, I'll ask you to take a look at my PR and make sure it works for you.